### PR TITLE
Add analytics API and documentation.

### PR DESCRIPTION
**Pull request by hokyungh**
### Description of changes:
This is to add some basic analytics. 

### Testing done:
Verified that each event is emitted.

#### Manual test cases (add more as needed):

- [x] Join meeting without CallKit
- [ ] Join meeting with CallKit as Incoming
- [ ] Join meeting with CallKit as Outgoing
- [X] Leave meeting
- [X] Rejoin meeting
- [X] See metrics
- [X] See attendee presence
- [X] Send audio
- [X] Receive audio
- [X] Mute self
- [X] Unmute self
- [X] See local mute indicator when muted
- [X] See remote mute indicator when other is muted
- [X] Enable and disable local video
- [X] Enable and disable remote video from remote side
- [X] Send local video
- [ ] Pause and resume remote video
- [X] Switch local camera
- [ ] Receive remote screen share

#### Integration validation (required before release)

- [X] Unit tests pass
- [ ] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [ ] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [ ] Test Demo against simulator with SDK (debug build) in workspace
- [ ] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

### Screenshots, if available:

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AmazonChimeSDK/AmazonChimeSDK/analytics/DefaultEventAnalyticsController.swift [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+//  DefaultEventAnalyticsController.swift
+//  AmazonChimeSDK
+//
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+@objcMembers public class DefaultEventAnalyticsController: EventAnalyticsController {
```
Comment (zhinang-amazon) [inline AmazonChimeSDK/AmazonChimeSDK/analytics/DefaultEventAnalyticsController.swift:11]: NSObject

File: AmazonChimeSDK/AmazonChimeSDK/internal/analytics/DefaultMeetingStatsCollector.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+//  DefaultMeetingStatsCollector.swift
+//  AmazonChimeSDK
+//
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+@objcMembers public class DefaultMeetingStatsCollector: MeetingStatsCollector {
```
Comment (zhinang-amazon) [inline AmazonChimeSDK/AmazonChimeSDK/internal/analytics/DefaultMeetingStatsCollector.swift:11]: NSObject

File: AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift [hunk 3]
```diff
@@ -27,23 +27,32 @@ class DefaultVideoClientController: NSObject {
     // This internal camera capture source is used for `startLocalVideo()` API without parameter.
     private let internalCaptureSource: DefaultCameraCaptureSource
     private var isInternalCaptureSourceRunning = true
+    private var eventAnalyticsController: EventAnalyticsController
```
Comment (zhinang-amazon) [inline AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift:30]: let?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,18 @@
 ## Unreleased
 
+### Added
+* Added Analytics
+    * `EventAnalyticsController`, `EventAnalyticsFacade`, `EventAnalyticsObserver` to handle analytics.
+    * Added `EventAttributesName`, `EventName` for meeting event information.
+    * **Breaking** Added `EventAnalyticsController` to parameters of `DefaultAudioVideoFacade`.
+    * Added `EventAnalyticsController` to parameters of `DefaultCameraCaptureSource`, `DefaultAudioClientObserver`, `DefaultAudioClientController`.
+    * Added `externalMeetingId` to property of `MeetingSessionConfiguration`.
+    * Added `EventAnalyticsController` to property of `MeetingSession`.
```
Comment (zhinang-amazon) [inline CHANGELOG.md:10]: This is a breaking change as well, right?

**Review by zhinang-amazon — DISMISSED** [review]

**Review by linsang21 — APPROVED** [review]

**Review by zhinang-amazon — APPROVED** [review]

**Pull request merged**